src/UseCases/CancelDonation/CancelDonationUseCase.php

```diff
@@ -49,7 +58,7 @@ public function cancelDonation( CancelDonationRequest $cancellationRequest ): Ca
 
 		$this->sendConfirmationEmail( $donation );
```

Comment:
Does a "your donation was cancelled" message not get shown anyway? If so, I'm not sure what the added value is of mentioning the mail confirmation failed. Cause people that already read the confirmation on the page don't really need the mail to begin with no? Anyway, please do add a TODO if you think this needs further attention.

There certainly is a more general TODO with regard to transaction handling. Perhaps not in this UC, but I figure there are cases where when something fails after the DB write, such as sending a critical email, the whole UC fails and the write gets rolled back (or not committed in the first place).